pinot-core/src/main/java/org/apache/pinot/core/plan/CombinePlanNode.java

```diff
@@ -190,6 +194,13 @@ public List<Operator> callJob() {
         return new SelectionOnlyCombineOperator(operators, _queryContext, _executorService);
       } else {
         // Selection order-by
+        if (QueryContextUtils.isMinMaxBasedSelectionOrderBy(_queryContext)) {
```

Comment:
IMO no need to put it into `QueryContextUtils`. This logic should be within the combine plan